src/test/ui/static/static-reference-to-fn-2.stderr

```diff
@@ -37,7 +37,10 @@ error[E0515]: cannot return value referencing temporary value
 LL | /     StateMachineIter {
 LL | |
 LL | |         statefn: &id(state1 as StateMachineFunc)
-   | |                   ------------------------------ temporary value created here
+   | |                   ------------------------------
+   | |                   |
+   | |                   temporary value created here
+   | |                   help: Use `.collect()` to allocate the iterator
```

Comment:
Given that the diagnostic can trigger for this test, we probably don't want the suggestion to be `MachineApplicable`.

Reply:
I changed it to `MaybeIncorrect`